### DISCUSSION TITLE
Allow conversion to JsonObject to be optional during amqp message deserialization

### DISCUSSION THREAD

**Discussion by ylepikhov** [smallrye/smallrye-reactive-messaging#2747] [Q&A] [answered]
Documentation [states](https://smallrye.io/smallrye-reactive-messaging/latest/amqp/receiving-amqp-messages/#deserialization):
`The connector converts incoming AMQP Messages into Reactive Messaging Message<T> instances. T depends on the body of the received AMQP Message.`
In case of `AMQP Data (with binary content) and the content-type is set to application/json` ***T*** becomes [JsonObject](https://vertx.io/docs/apidocs/io/vertx/core/json/JsonObject.html).
It might be unwanted. One might want to leave payload as is (without any deserialization/serialization), or use alternative deserialization (Jackson, etc).
It will be nice to have a configuration option for AmqpConnetor to disable conversion to JsonObject in case of json content type.

**Comment by ozangunalp** [comment, marked as answer]
In that case, you can inject `io.vertx.amqp.AmqpMessage` to the application and unwrap it to access the low level Proton API `Message`. If you'd like to handle it transparently to the consuming methods, you can implement a `MessageConverter`.
